### PR TITLE
Part 2 of #36878

### PR DESCRIPTION
**Pull request by nikosdion**
Fixed the missing calls to opcache file invalidation as per the discussion in #36878 

/cc @zero-24 @richard67

### CODE REVIEW

File: administrator/components/com_joomlaupdate/restore.php [hunk 1]
```diff
@@ -1438,6 +1438,10 @@ public function process()
 		{
 			@touch($this->filename, $this->timestamp);
 		}
+		if (substr($this->filename, -4) === '.php')
+		{
+			$this->clearFileInOPCache($this->filename);
+		}
```
Comment (richard67) [inline administrator/components/com_joomlaupdate/restore.php:1445]:
```suggestion

		if (substr($this->filename, -4) === '.php')
		{
			$this->clearFileInOPCache($this->filename);
		}
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 9fd4766dd255b50e187f8bd430b1ac8110f48ead<br /><br />I've tested lots of updates on that environment where I most of the times can reproduce the issue and could also today.

With the PR applied before the update, the issue did not happen anymore.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37003">issues.joomla.org/tracker/joomla-cms/37003</a>.</sub>

**Comment by richard67** [comment]
@nikosdion Do you think that a similar change in the J4 extract.php could make sense? I don't see there that we do there what this PR here does. Or is that just at another place?

**Comment by nikosdion** [comment]
Joomla 4's `extract.php` already does that. The benefits of creating a purpose-built extraction feature instead of reusing generic code that's really written for restoring backups on empty servers ;)

**Comment by richard67** [comment]
> Joomla 4's `extract.php` already does that. The benefits of creating a purpose-built extraction feature instead of reusing generic code that's really written for restoring backups on empty servers ;)

@nikosdion Hmm, I did not see we do it on all extracted PHP files. But maybe I've just missed that when having looked quickly into the code.

**Comment by nikosdion** [comment]
@richard67 Look at `ZIPExtraction::processLastExtractedFile` which is called after we're done extracting a file. It calls `clearFileInOPCache` which calls `opcache_invalidate` for files with a .php extension. I think that you can follow the code backwards and reference the (documented!) constants to understand how it works.

**Comment by richard67** [comment]
> @richard67 Look at `ZIPExtraction::processLastExtractedFile` which is called after we're done extracting a file. It calls `clearFileInOPCache` which calls `opcache_invalidate` for files with a .php extension. I think that you can follow the code backwards and reference the (documented!) constants to understand how it works.

@nikosdion Thanks. Found it. Seems indeed to be ok.

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on 9fd4766dd255b50e187f8bd430b1ac8110f48ead<br /><br />I have just tested on my setup too so I think we are good to go here. Thanks @nikosdion :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37003">issues.joomla.org/tracker/joomla-cms/37003</a>.</sub>

**Comment by richard67** [comment]
Yes, I tested again today to make sure it also works on odd and not only even days on my weird environment, and it was all fine 😄 .

**Pull request merged**